### DISCUSSION TITLE
Is it possible to create a new markdown render in the future?

### DISCUSSION THREAD

**Discussion by Samtoto** [thephpleague/commonmark#644] [Q&A] [answered]
**Description**  
Is it possible to create a new markdown render in the future?  
Consider such a situation:  
The user wants to edit the submitted markdown document, not the parsed html document.  

**Example**  
```php
    $markdown = <<<EOT
    ```html
    <?php
    class A() {}
    ?>
    <script>alert('safe')</script>
    ```
    <script>alert('XSS')</script>
    EOT;

    $commonMask = new CommonMarkConverter(
        ['html_input' => 'strip', 'allow_unsafe_links' => false],
    );

    $cleanedMarkdown = $commonMask->convertToMarkdown($markdown);
    echo $cleanedMarkdown; // String
    // Expect output:

    '''
    ```html
    &lt;?php
    class A() {}
    ?&gt;
    &lt;script&gt;alert('safe')&lt;/script&gt;
    ```
    '''
```

**Comment by Samtoto** [comment]
Found a similar [issue #141](https://github.com/thephpleague/commonmark/issues/141)

**Comment by colinodell** [comment, marked as answer]
Can you save the Markdown document in a database and then re-render it later when it is changed?

**Comment by Samtoto** [comment]
Yes, I changed my workflow, return the original content to owener, and return the parsed content to the other users.Thx a lot!
